### PR TITLE
latest work on reports

### PR DESCRIPTION
**Pull request by indirectlylit**
This work has the latest coach reports, including upgrades to the Topics and Recent views. It should be about ready to hook up to the exercise detail reports.

### CODE REVIEW

File: kolibri/core/assets/src/state/actions.js [hunk 1]
```diff
@@ -555,6 +555,9 @@ function createMasteryLog(store, masteryLevel, masteryCriterion) {
     totalattempts: 0,
     mastery_criterion: masteryCriterion,
   });
+  // Premptively set attributes
+  store.dispatch('SET_LOGGING_MASTERY_STATE', masteryLogModel.attributes);
```
Comment (rtibbles) [inline kolibri/core/assets/src/state/actions.js:559]: :+1:

File: kolibri/plugins/coach/assets/src/state/actions/reports.js [hunk 2]
```diff
@@ -77,21 +83,29 @@ function _showChannelList(store, classId) {
...
-    totalProgress: progressData.total_progress,
-    // content/learner reports
-    kind: progressData.kind,
-    nodeCount: progressData.node_count,
-    // 'recent' reports
-    logCountComplete: progressData.log_count_complete,
-    logCountTotal: progressData.log_count_total,
-  };
+function _contentReportState(data) {
+  console.log('dddddd', data);
```
Comment (rtibbles) [inline kolibri/plugins/coach/assets/src/state/actions/reports.js:87]: log

File: kolibri/plugins/coach/assets/src/state/actions/reports.js [hunk 3]
```diff
@@ -77,21 +83,29 @@ function _showChannelList(store, classId) {
...
+      nodeCount: progressData.node_count,
+      totalProgress: progressData.total_progress,
+    })),
+    title: row.title,
+  }));
 }
 
-function _reportState(data) {
+function _recentReportState(data) {
+  console.log('rrrrrrr', data);
```
Comment (rtibbles) [inline kolibri/plugins/coach/assets/src/state/actions/reports.js:108]: log

File: kolibri/plugins/coach/assets/src/state/actions/reports.js [hunk 4]
```diff
@@ -102,13 +116,33 @@ function _reportState(data) {
...
+      logCountComplete: progressData.log_count_complete,
+      logCountTotal: progressData.log_count_total,
+      totalProgress: progressData.total_progress,
+    })),
     title: row.title,
   }));
 }
 
+function _learnerReportState(data) {
+  console.log('lllllll', data);
```
Comment (rtibbles) [inline kolibri/plugins/coach/assets/src/state/actions/reports.js:129]: looooooooooggggggg

File: kolibri/plugins/coach/assets/src/state/actions/reports.js [hunk 5]
```diff
@@ -102,13 +116,33 @@ function _reportState(data) {
...
+      kind: progressData.kind,
+      timeSpent: progressData.time_spent,
+      totalProgress: progressData.total_progress,
+    })),
+  }));
+}
+
 function _contentSummaryState(data) {
   if (!data) { return {}; }
+  console.log('cscscscs', data);
```
Comment (rtibbles) [inline kolibri/plugins/coach/assets/src/state/actions/reports.js:145]: :laos:

Comment (rtibbles) [reply kolibri/plugins/coach/assets/src/state/actions/reports.js:145]: (just flagging these now)

File: kolibri/plugins/coach/assets/src/state/actions/reports.js [hunk 6]
```diff
@@ -120,121 +154,115 @@ function _contentSummaryState(data) {
...
+      nodeCount: progressData.node_count,
+      totalProgress: progressData.total_progress,
+    })),
     title: data.title,
   };
 }
 
 function _userSummaryState(data) {
-  // console.log('uuuuuu', data);
+  console.log('uuuuuu', data);
```
Comment (rtibbles) [inline kolibri/plugins/coach/assets/src/state/actions/reports.js:167]: :vietnam:

File: kolibri/plugins/coach/assets/src/state/actions/reports.js [hunk 7]
```diff
@@ -120,121 +154,115 @@ function _contentSummaryState(data) {
...
-  /* check if params are semi-valid. */
-  function _validate(value, constants) {
-    if (!values(constants).includes(value)) {
-      throw Error(`Invalid report parameters: ${value} not in ${JSON.stringify(constants)}`);
-    }
-  }
-  _validate(contentScope, ReportConstants.ContentScopes);
-  _validate(userScope, ReportConstants.UserScopes);
+function _setContentReport(store, reportPayload) {
+  console.log('?????', reportPayload);
```
Comment (rtibbles) [inline kolibri/plugins/coach/assets/src/state/actions/reports.js:175]: :cambodia:

File: kolibri/plugins/coach/assets/src/state/actions/reports.js [hunk 8]
```diff
@@ -120,121 +154,115 @@ function _contentSummaryState(data) {
...
+  const reportPromise = ContentReportResource.getCollection(reportPayload).fetch();
+  reportPromise.then(report => {
+    console.log('>>>>>', _contentReportState(report));
```
Comment (rtibbles) [inline kolibri/plugins/coach/assets/src/state/actions/reports.js:178]: :thailand:

File: kolibri/plugins/coach/assets/src/state/getters/reportUtils.js [hunk 9]
```diff
@@ -0,0 +1,69 @@
+
+const ReportConstants = require('../../reportConstants');
+const CoreConstants = require('kolibri.coreVue.vuex.constants');
+
+const ContentNodeKinds = CoreConstants.ContentNodeKinds;
+
+
```
Comment (rtibbles) [inline kolibri/plugins/coach/assets/src/state/getters/reportUtils.js:7]: These functions look ripe for unit testing.

File: kolibri/plugins/coach/assets/src/state/getters/reportUtils.js [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+/* given an array of objects sum the keys on those that pass the filter */
+function _sumOfKeys(array, key, filter = () => true) {
```
Comment (rtibbles) [inline kolibri/plugins/coach/assets/src/state/getters/reportUtils.js:9]: I notice that this function is only used twice in the foregoing - a slight pre-abstraction, perhaps?

File: kolibri/plugins/coach/assets/src/state/store.js [hunk 11]
```diff
@@ -18,8 +19,26 @@ const mutations = {
 
   // report
   SET_REPORT_SORTING(state, sortColumn, sortOrder) {
-    state.pageState.sortColumn = sortColumn;
-    state.pageState.sortOrder = sortOrder;
+    Vue.set(state.pageState, 'sortColumn', sortColumn);
+    Vue.set(state.pageState, 'sortOrder', sortOrder);
+  },
+  SET_REPORT_PROPERTIES(state, options) {
+    Vue.set(state.pageState, 'classId', options.classId);
```
Comment (rtibbles) [inline kolibri/plugins/coach/assets/src/state/store.js:26]: Vue.set :all-the-things:

**Comment by rtibbles** [comment]
Tested manually, does what I expect. Clean up can happen on detail-reports.

File: kolibri/plugins/coach/assets/src/state/getters/reportUtils.js (hunk 9 above)
Comment (indirectlylit) [reply kolibri/plugins/coach/assets/src/state/getters/reportUtils.js:7]: agree.

Note that most of these functions pre-dated the current refactor, and I'm considering removing some...

File: kolibri/plugins/coach/assets/src/views/index.vue [hunk 12]
```diff
@@ -78,13 +85,15 @@
...
+        return this.pageName !== Constants.PageNames.CLASS_LIST && (this.isCoach || this.isAdmin);
       },
     },
     vuex: {
       getters: {
         pageName: state => state.pageName,
-        isCoachAdminOrSuperuser,
+        isSuperuser: coreGetters.isSuperuser,
+        isAdmin: coreGetters.isAdmin,
+        isCoach: coreGetters.isCoach,
```
Comment (indirectlylit) [inline kolibri/plugins/coach/assets/src/views/index.vue:96]: I want to use these new getters for #1250 and #1238

**Pull request merged**